src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -545,6 +554,9 @@ internal enum ExceptionResource
         ExpectedJsonTokens,
         TrailingCommaNotAllowedBeforeArrayEnd,
         TrailingCommaNotAllowedBeforeObjectEnd,
+        InvalidCharacterAtStartOfComment,
+        ExpectedComment,
+        UnexpectedEndOfDataWithinStartOfComment
```

Comment:
@baronjs, I would appreciate a second set of eyes here, especially around exception messages and when they are being thrown :) Thanks!

Reply:
So essentially I have removed all the three above and replaced them with `UnexpectedEndOfDataWhileReadingComment` which maps to `Unexpected end of data while reading a comment.` as suggested by @bartonjs over [here](https://github.com/dotnet/corefx/pull/35934#discussion_r281215881).

@ahsonkhan please check and let me know if this looks good. Thanks!